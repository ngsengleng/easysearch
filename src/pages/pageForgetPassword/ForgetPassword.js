import React, { useEffect } from "react";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import TextField from "@material-ui/core/TextField";
import Link from "@material-ui/core/Link";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";

import { firebase } from "@firebase/app";
import { useForm, Controller } from "react-hook-form";
import { useHistory } from "react-router-dom";

function Copyright() {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {"Copyright Â© "}
      <Link color="inherit" href="/">
        EasySearch
      </Link>{" "}
      {new Date().getFullYear()}
      {"."}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

export default function SignIn() {
  const classes = useStyles();
  const { handleSubmit, control } = useForm();
  const history = useHistory();
  const onSubmit = async (data) => {
    firebase
      .auth()
      .sendPasswordResetEmail(data.Email)
      .then(() => {
        alert("email has been sent!");
        history.push("/");
      })
      .catch((error) => {
        const errorCode = error.code;
        if (errorCode === "auth/user-not-found") {
          alert("This user does not exist.");
        }
      });
  };

  useEffect(() => {
    document.title = "Login";
  }, []);

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Typography component="h1" variant="h5">
          Reset password
        </Typography>
        <form className={classes.form} onSubmit={handleSubmit(onSubmit)}>
          <Controller
            name="Email"
            control={control}
            defaultValue=""
            render={({ field: { onChange, value }, fieldState: { error } }) => (
              <TextField
                variant="outlined"
                margin="normal"
                fullWidth
                id="email"
                label="Email Address"
                autoComplete="email"
                autoFocus
                value={value}
                onChange={onChange}
                error={!!error}
                helperText={error ? error.message : null}
              />
            )}
            rules={{ required: "Email required" }}
          />
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            Reset password
          </Button>
          <Grid container>
            <Grid item xs>
              <Link href="/" variant="body2">
                Back to login
              </Link>
            </Grid>
          </Grid>
        </form>
      </div>
      <Box mt={8}>
        <Copyright />
      </Box>
    </Container>
  );
}
