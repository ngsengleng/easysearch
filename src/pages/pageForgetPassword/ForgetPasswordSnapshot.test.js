import React from "react";
import renderer from "react-test-renderer";
import ForgetPassword from "./ForgetPassword";
import { AuthProvider } from "../../context/AuthContext";
it("renders correctly", () => {
  const tree = renderer
    .create(
      <AuthProvider>
        <ForgetPassword />
      </AuthProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
