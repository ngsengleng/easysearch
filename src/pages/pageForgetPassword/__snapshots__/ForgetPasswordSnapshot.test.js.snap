// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Reset password
    </h1>
    <form
      className="makeStyles-form-3"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="email"
          id="email-label"
        >
          Email Address
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="email"
            autoFocus={true}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-7"
            >
              <span>
                Email Address
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Reset password
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="/"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Back to login
          </a>
        </div>
      </div>
    </form>
  </div>
  <div
    className="MuiBox-root MuiBox-root-9"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright Â© 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        EasySearch
      </a>
       
      2021
      .
    </p>
  </div>
</main>
`;
